safe/impact_functions/generic/classified_polygon_building/impact_function.py

```diff
@@ -0,0 +1,202 @@
+# coding=utf-8
+"""InaSAFE Disaster risk tool by Australian Aid - Generic Polygon on Building
+Impact Function.
+
+Contact : ole.moller.nielsen@gmail.com
+
+.. note:: This program is free software; you can redistribute it and/or modify
+     it under the terms of the GNU General Public License as published by
+     the Free Software Foundation; either version 2 of the License, or
+     (at your option) any later version.
+
+"""
+
+from collections import OrderedDict
+
+from safe.storage.vector import Vector
+from safe.utilities.i18n import tr
+from safe.impact_functions.base import ImpactFunction
+from safe.impact_functions.generic.classified_polygon_building\
+    .metadata_definitions import ClassifiedPolygonBuildingFunctionMetadata
+from safe.common.exceptions import InaSAFEError
+from safe.common.utilities import (
+    get_thousand_separator,
+    get_non_conflicting_attribute_name,
+    get_osm_building_usage,
+    color_ramp)
+from safe.engine.interpolation import (
+    assign_hazard_values_to_exposure_data)
+from safe.impact_reports.building_exposure_report_mixin import (
+    BuildingExposureReportMixin)
+
+
+class ClassifiedPolygonBuildingFunction(
+        ImpactFunction,
+        BuildingExposureReportMixin):
+    """Impact Function for Generic Polygon on Building."""
+
+    _metadata = ClassifiedPolygonBuildingFunctionMetadata()
+
+    def __init__(self):
+        super(ClassifiedPolygonBuildingFunction, self).__init__()
+
+        # Hazard zones are all unique values from the hazard zone attribute
+        self.hazard_zones = []
+
+    def notes(self):
```

Comment:
By validated I mean that you will get an exception if you try to reference an invalid property of the note etc...